parsers/IN_PB.py

```diff
@@ -84,6 +78,30 @@ def fetch_production(zone_key='IN-PB', session=None, target_datetime=None, logge
     return data
 
 
+def read_punjab_consumption_date(date_text, time_text, current):
+    date_time = date_text + time_text + 'Asia/Kolkata'
+
+    dates = []
+    try:
```

Comment:
can you add a few comments as to what you're doing here?
Will be easier to maintain in the future!